File: Python/Easy/First Non-Repeating Character/solution_2.py
def firstNonRepeatingCharacter(string):

    frequency = {}

    for element in string:
        frequency[element] = frequency.get(element, 0) + 1

    for index in range(len(string)):
        letter = string[index]
        if frequency[letter] == 1:
            return index
            
    return -1

File: Go/Easy/First Non-Repeated Character/solution_2.go
package main

// O(n) time / O(1) space since only 26 letters it can be
func FirstNonRepeatingCharacter(str string) int {

	frequency := make(map[rune]int)

	for _, letter := range str {
		frequency[letter] = frequency[letter] + 1
	}

	for index, letter := range str {
		if frequency[letter] == 1 {
			return index
		}
	}

	return -1
}
